File: Low Level Design (LLD)/Problems/07. SocialApp/Cpp/src/Comment.cpp
#include "Comment.h"
#include "User.h"

Comment::Comment(int id, std::shared_ptr<User> author, const std::string& content)
    : id_(id), author_(author), content_(content) {}

int Comment::getId() const { return id_; }
std::shared_ptr<User> Comment::getAuthor() const { return author_; }
const std::string& Comment::getContent() const { return content_; }

File: Low Level Design (LLD)/Problems/07. SocialApp/Java/models/Comment.java
package models;

public class Comment {
    private final int id;
    private final User author;
    private final String content;

    public Comment(int id, User author, String content) {
        this.id = id;
        this.author = author;
        this.content = content;
    }

    // Getters
    public int getId() {
        return id;
    }

    public User getAuthor() {
        return author;
    }

    public String getContent() {
        return content;
    }
}
